### ISSUE TITLE
Reduce networking and ota boiler plate

### ISSUE THREAD

**Issue by cookejames** [randuhmm/rad-esp8266#2]
https://github.com/randuhmm/rad-esp8266/blob/master/examples/TestExample/TestExample.ino

Your examples include a lot of networking and OTA boiler plate that have to be implemented in each sketch. It would be useful to move all this to the RADConnector or a new networking class.

**Comment by randuhmm** [comment]
I purposely left out a lot of these setup and configuration features from the library for a few reasons:

- Users might not want OTA enabled for security reasons. Giving anyone the ability to push firmware to a device if they are on the same network can be concerning to some (of course, you could argue that they could password protect the OTA).
- The nature of the WiFi setup and config is somewhat outside the scope of this library. What I mean is that users should be free to connect to a WiFi network in any way they see fit; this library only presumes that by the time it is initialized, a network connection is established. Some ways a user might make a network connection include:
    - Hard-coding the SSID and password
    - Using the WiFiManager library to create a configuration AP (hotspot) to allow a user to set and reset the WiFi credentials
    - Allow for a Serial connected NFC reader device that you can tap with the credentials to set or change the WiFi (does not exist... yet :P)
   - Some other PTP way of sending WiFi access credentials. I have seen routers that have a "simple connect button" to easily allow a device to join the network, but I don't know much about how the credentials are passed along.

I do agree that the boilerplate is unsightly, but I also don't want this library to become too opinionated with how users can and cannot make their network connection. I might be open to some form of encapsulation like you mentioned with a separate "networking class", but the value to the user needs to be apparent. Perhaps, looking ahead to ESP-32, this type of encapsulation is how multiple platforms might be supported from this code-base in the future.

**Comment by cookejames** [comment]
I agree but it doesn't hurt to provide an easy default. If you haven't tried it you should give @erocm123's sonoff integration a go. Its super easy - simply flash a bin to your ESP and install an app and DH. You basically hit discover and bam it finds and configures your devices.

I would love to provide that ease of use by default without restricting what people can add themselves. A very simple network interface with a default implementation that uses wifimanager and ota would be great. People can then opt not to use it if they want to write their own but I'm sure that most users just want to get into writing cool integrations with their ESP and avoid as much boilerplate as they can.

**Comment by randuhmm** [comment]
Yea, I really like the idea of releasing pre-built binaries along with a circuit diagram (or a link to a schematic + OSH Park PCB + BOM). I have several friends and relatives who are interested in experimenting with IoT and I would like to help them out with some cost-effective solutions that they can tinker with themselves. 

I think our best option to easily skip the messy boiler-plate is to allow for an optional include to handle all the AutoConnect features. This will simplify the sketches for those who are not interested in customizing the WiFi connection and credential negotiation but not hamstring those who want to use their own methods.
